### DISCUSSION TITLE
MaskedBehavior (Regex)

### DISCUSSION THREAD

**Discussion by L3Prichaun13** [CommunityToolkit/Maui#516] [New Feature Discussions]
Very nice work on the current set of MaskedBehaviors.... Can you guys add one for Regex? This would GREATLY increase how powerful your Mask class is. Right now I can only mask data size but I cannot handle special situations where I want to force the user to type 2 chars followed by 6 numbers. If I could mask with regex, then pretty much every scenario under the sun would be covered.
I would much rather use this toolkit instead of others like Telerik or DevExpress of Infragistics

**Comment by TheCodeTraveler** [comment]
That'd be cool!

I'm admittedly not an expert in RegEx. 

Do you happen to have a prototype that shows off we could implement it?

**Reply by L3Prichaun13** [reply]
I do not at the moment.. I know Telerik has this option but I would love to get it into here with its own code set of course.
I will see if I can figure anything out to help but hopefully someone on your team can also do the same :)

**Reply by L3Prichaun13** [reply]
There is also a ton of documentation online about regex.... I don't think you need to know too much.... Mostly just need to be able to make use of the MS Regex class by pulling in the regex from the mask then applying it to the value in the entry control and have a way for characters not found in that regex be removed and the updated value posted back to the entry control. I did a quick search and found many articles around this for various languages. Below is one of those articles
https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex?view=net-6.0

**Reply by IeuanWalker** [reply]
This may help
- https://mudblazor.com/features/masking#regexmask-regexmask.email
- https://github.com/MudBlazor/MudBlazor/blob/dev/src/MudBlazor/Utilities/MaskAlgorithms/RegexMask.cs

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
